eZ/Publish/Core/FieldType/RichText/Resources/schemas/docbook/docbook.iso.sch

```diff
@@ -246,6 +246,7 @@
     <s:rule context="db:link">
       <s:assert test="not(.//db:link)">link must not occur in the descendants of link</s:assert>
       <s:assert test="not(.//db:ezlink)">ezlink must not occur in the descendants of link</s:assert>
+      <s:assert test="not(contains(@*[name()='xlink:href'], 'javascript:') or contains(@*[name()='xlink:href'], 'vbscript:'))">using scripts in links is not allowed</s:assert>
```

Comment:
Correct, URLs are handled in `RichTextStorage`, whose gateways inherits from `Url` field type's gateway. So the solution would be to validate URLs for XSS when storing them into common URL storage. Maybe this would be a good opportunity to handle it properly and finally implement `UrlService`. I probably already mentioned this at some point, also hinted here:

- https://github.com/ezsystems/ezpublish-kernel/blob/e2d26faa54715675fc8f1eef3defbe192973bf5b/eZ/Publish/Core/FieldType/RichText/Converter/Render/Embed.php#L196
- https://github.com/ezsystems/ezpublish-kernel/blob/e2d26faa54715675fc8f1eef3defbe192973bf5b/eZ/Publish/Core/FieldType/RichText/Converter/Link.php#L123